sphinx_gallery/directives.py

```diff
@@ -67,34 +96,67 @@ def run(self):
             lines.append(heading_level * len(heading))
 
         def has_backrefs(obj):
-            src_dir = config.sphinx_gallery_conf["src_dir"]
-            path = os.path.join(src_dir, backreferences_dir, f"{obj}.examples")
-            return os.path.isfile(path) and os.path.getsize(path) > 0
+            path = Path(src_dir, backreferences_dir, f"{obj}.examples")
+            return path if (path.is_file() and (path.stat().st_size > 0)) else False
+
+        file_paths = []
+        for obj in obj_list:
+            if path := has_backrefs(obj):
+                file_paths.append((obj, path))
+            elif paths := Path(src_dir).glob(obj):
+                file_paths.extend([(obj, p) for p in paths])
 
-        if not any(has_backrefs(obj) for obj in obj_list):
+        if len(file_paths) == 0:
             return []
 
-        # Insert the backreferences file(s) using the `include` directive.
-        # This already includes the opening <div class="sphx-glr-thumbnails">
-        # and its closing </div>.
-        for obj in obj_list:
-            path = os.path.join(
-                "/",  # Sphinx treats this as the source dir
-                backreferences_dir,
-                f"{obj}.examples",
-            )
-
-            # Always remove the heading from the file
-            lines.append(
-                f"""\
-.. include:: {path}
-    :start-after: start-sphx-glr-thumbnails"""
-            )
-
-        # Parse the assembly of `include` and `raw` directives
+        lines.append(THUMBNAIL_PARENT_DIV)
+
+        # sort on the str(file_path) but keep (obj, path) pair
+        sortkey = config.sphinx_gallery_conf["minigallery_sort_order"]
+        for obj, path in sorted(
+            set(file_paths),
+            key=((lambda x: sortkey(str(x[-1].name))) if sortkey else None),
+        ):
+            if path.suffix == ".examples":
+                # Insert the backreferences file(s) using the `include` directive.
+                # / is the src_dir for include
+                lines.append(
+                    f"""\
+.. include:: /{path.relative_to(src_dir).as_posix()}
+    :start-after: thumbnail-parent-div-open
+    :end-before: thumbnail-parent-div-close"""
+                )
+            else:
+                dirs = [
+                    (e, g)
+                    for e, g in zip(
+                        config.sphinx_gallery_conf["examples_dirs"],
+                        config.sphinx_gallery_conf["gallery_dirs"],
+                    )
+                    if (obj.find(e) != -1)
+                ]
+                if len(dirs) != 1:
+                    raise ExtensionError(
+                        f"Error in gallery lookup: input={obj}, matches={dirs}, "
+                        f"examples={config.sphinx_gallery_conf['examples_dirs']}"
+                    )
```

Comment:
@story645 I stumbled on this error accidentally but I don't fully follow what situation this is meant to catch?

I managed to contrive a case where I added a file to a path not in `examples_dir` to a minigallery directive. In that case there were no matches and the error message didn't really make sense.

I assume it is meant to catch cases when `dirs>1`? So maybe a sub-gallery directory is named the same as a examples gallery?

Reply:
A more informative error message would be great. There is no test for this but depending on what this is mean to catch, maybe it would be too difficult to write a test.

Reply:
cc @larsoner just in case you remember...? :grimacing:

Reply:
No idea :disappointed: